dartagnan/src/main/java/com/dat3m/dartagnan/program/event/EventFactory.java

```diff
@@ -106,12 +106,12 @@ public static Store newStoreWithMo(Expression address, Expression value, String
         return store;
     }
 
-    public static Fence newFence(String name) {
-        return new Fence(name);
+    public static GenericVisibleEvent newFence(String name) {
+        return new GenericVisibleEvent(name, Tag.FENCE);
```

Comment:
I would return `GenericVisibleEvent(name, name, Tag.FENCE)` to make it semantically equivalent with the previous version. Then you do not need to change any of the other fence factory methods.

Reply:
That would add `name` to the tag to every single event which is exactly what you initially wrote we should not do. I am having doubts about what would be the best approach. On the one side, I think it is better not to add the name by default and do it only if needed by creating a specific factory method for the special instruction and add the tag there (in like we are doing now in e.g., RISCV). However, except for PTX (and the soon coming vulkan), it seems pretty much all our memory models indeed require having the name as a tag.

Reply:
I said that it would not be a good idea to do it in the constructor of `GenericVisibleEvent` and that the `EventFactory` could do it, if necessary. Not doing it in the constructor allows you to easily support `EventFactory.newFence` which automatically adds name + F-tag, while also supporting something like `EventFactory.newVisibleEvent` which does not add anything at all. Honestly, we might even go a step further and just have `GenericEvent` that does not even need to be visible and let the `EventFactory` add the visible tag.
That being said, I understood that `PTX` and `Vulkan` just do not use the F-tag, so they have non-fence events.
So you wouldn't call `EventFactory.newFence` in the first place, whether it adds the name or not.

Reply:
I am not referring to the new instructions from vulkan that gave raise to the change in this PR, but rather to PTX or vulkan **fences**. E.g. PTX has [this kind](https://github.com/hernanponcedeleon/Dat3M/blob/027b4024dd3b9894473233bc5f197912a7cf7421/litmus/PTX/Proxy-Alias-AliasFence.litmus#L10) of instructions which in the memory model are simply referred as `F` (different to e.g. `tso` which expliclty uses `mfence`)

Reply:
I see. Well, that is also the case for all C11 fences: the cat model always refers to `F & mo` never to the actual fence name which would be `atomic_thread_fence` I suppose. So there a few cases where the name of the fence is irrelevant.

Reply:
Exactly and this is why I don't want `newFence` to add it. Instead, we should call e.g., `newSYBarrier` and that one would add the needed tag.

Reply:
Ok, but you could still add a method like `newFenceWithoutNameTag` (you can even make it private if you want) to avoid the code repetition. I mean, we can easily provide different options with different factory methods.
Though I guess in most cases it does not really hurt to add the name as a tag as well, but I agree that we can avoid it entirely.

Reply:
It seems pretty much all our MMs were relying on the name being add as a tag, either because the cat file refers to it (e.g., `mfence`) or because we about this and use it to pass the memory ordering (e.g., compilation of C11). Thus, I decided to keep the functionality as closed as possible to before.